ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/MessageDrivenComponentDescriptionFactory.java

```diff
@@ -118,34 +118,17 @@ private void processMessageBeans(final DeploymentUnit deploymentUnit, final Coll
                 beanClassName = override(beanClassInfo.name().toString(), beanMetaData.getEjbClass());
                 deploymentDescriptorEnvironment = new DeploymentDescriptorEnvironment("java:comp/env/", beanMetaData);
 
-                if (beanMetaData instanceof MessageDrivenBeanMetaData) {
-                    //It may actually be GenericBeanMetadata instance
-                    final MessageDrivenBeanMetaData mdb = (MessageDrivenBeanMetaData) beanMetaData;
-                    messagingType = mdb.getMessagingType();
-                    final ActivationConfigMetaData activationConfigMetaData = mdb.getActivationConfig();
-                    if (activationConfigMetaData != null) {
-                        final ActivationConfigPropertiesMetaData propertiesMetaData = activationConfigMetaData.getActivationConfigProperties();
-                        if (propertiesMetaData != null) {
-                            for (final ActivationConfigPropertyMetaData propertyMetaData : propertiesMetaData) {
-                                activationConfigProperties.put(propertyMetaData.getKey(), propertyMetaData.getValue());
-                            }
+                final MessageDrivenBeanMetaData mdb = beanMetaData;
+                messagingType = mdb.getMessagingType();
```

Comment:
looks like we can directly use the existing variable `beanMetaData`, without creating the new local variable `mdb`.

Reply:
Fixed.